### ISSUE TITLE
MiniTV not working on LilyGo T-Display ESP32-S3 yet

### ISSUE THREAD

**Issue by richgain** [moononournation/MiniTV#7]
Hi

I love this project and I have been trying hard to get the code running on a LilyGo T-Display ESP32-S3 with attached SD card and amplifier. 

The hardware is all working now and the code is almost there, but the picture is broken into a patchwork of blocks, some recognisable and others skewed diagonally. 

My source code is on GitHub here - https://github.com/richgain/MiniTV-on-T-Display-ESP32S3

And the video of my current output is on YouTube here - https://youtu.be/X-qEaaX9PE0?si=JbeYytWCOqluqvPf 
Note, this video is longer and clearer than the one on my GitHub page. 

I would really appreciate some help with what needs changing to make this device work correctly. Thanks.

**Comment by moononournation** [comment]
Seems you have not yet tested the display class declaration. I recommend you test the Arduino_GFX PDQgraphicstest example first.

**Comment by richgain** [comment]
![IMG_1255](https://github.com/moononournation/MiniTV/assets/1440566/66311777-d35a-4211-9c42-d015e00b840e)
Yes, the Arduino_GFX PDQgraphicstest example was the first program I tried and it runs perfectly. 
This is what I get. 
https://youtube.com/shorts/uLH8X2pptLg?si=4jKDU2G7fG0vBHpg

**Comment by richgain** [comment]
This is what I get.

**Comment by moononournation** [comment]
then why you use Arduino_ESP32PAR8Q class?

**Comment by richgain** [comment]
I'm not sure. I am not really a programmer - just learning by reading code.
I thought I was copying the display and databus from the PDQGraphics test to the MiniTV.ino.

![image](https://github.com/moononournation/MiniTV/assets/1440566/28e3d195-eebf-47a8-b54c-009468a7eb43)

**Comment by moononournation** [comment]
may be you can try aviPlayer Cinepak example:
https://github.com/moononournation/aviPlayer/tree/main/AviMp3Cinepak

**Comment by richgain** [comment]
Thanks. I'll take a look at it.

**Comment by richgain** [comment]
No luck so far.
When I replace the Arduino_GFX setting block with the settings for LILYGO_T_DISPLAY_S3, and try to compile it, I get an error message:

C:\Users\richg\Documents\Arduino\ESP32\AviMp3Cinepak\AviMp3Cinepak.ino: In function 'void setup()':
C:\Users\richg\Documents\Arduino\ESP32\AviMp3Cinepak\AviMp3Cinepak.ino:136:25: error: 'class Arduino_GFX' has no member named 'getFramebuffer'
       output_buf = gfx->getFramebuffer();
                         ^~~~~~~~~~~~~~
exit status 1

Compilation error: 'class Arduino_GFX' has no member named 'getFramebuffer'

**Comment by moononournation** [comment]
sorry, you should use https://github.com/moononournation/aviPlayer/tree/main/AviMp3CinepakDMA

**Comment by richgain** [comment]
Thank you, this works!
My first attempt played the video sideways across the left hand edge of the screen. 
Then I realised that the T-Display S3 is basically designed as a 'portrait-mode' device. 
Setting the rotation to 1, instead of 0, gives the correct aspect but the video plays very, very slowly. 

In the end, I discovered that the best solution is to rotate the source video 90° clockwise using Handbrake software, and then encode the rotated version as width 168, height 320, using ffmpeg. 
This looks great on the screen and plays at full speed.
